client/collections/mixins.coffee

```diff
@@ -10,18 +11,18 @@ _ = require 'underscore'
     { @total, @count } = data
     data.results
 
-  getOrFetchIds: (ids, options = {}) ->
-    ids = _.without ids, @pluck('id')...
-    ids = _.reject ids, (id) -> not id
-    return options.success? this unless ids.length
-    @fetch
-      data: ids: ids
-      error: options.error
-      remove: false
-      success: options.success
-      complete: options.complete
-
 @Filter =
 
   notIn: (col) ->
-    @reject (model) => model.id in col.pluck 'id'
\ No newline at end of file
+    @reject (model) => model.id in col.pluck 'id'
+
+  getOrFetchIds: (ids, options = {}) ->
+    async.map ids or [], (id, cb) =>
+      new @model(id: id).fetch
+        error: (m, err) -> cb null, err
+        success: (model) -> cb null, model
+    , (err, models) =>
+      options.complete?()
```

Comment:
I refactored `getOrFetchIds` to DRY it up into the mixin. I think the way async was being used in tandem with options was causing the `complete` callback not get called properly, so I moved the callbacks until after all of the fetched/failed calls.